app/views/users/_form.html.erb

```diff
@@ -0,0 +1,32 @@
+<%= form_with(model: user, local: true) do |form| %>
+  <% if user.errors.any? %>
+    <div id="error_explanation">
+      <h2><%= pluralize(user.errors.count, "error") %> prohibited this user from being saved:</h2>
+
+      <ul>
+        <% user.errors.full_messages.each do |message| %>
+          <li><%= message %></li>
+        <% end %>
+      </ul>
+    </div>
+  <% end %>
```

Comment:
* [ ] **_There should be NO application logic in the views (no `Post.new`, `Comment.create`, `current_user.friends.find_by(...)`...). Everything connected with data processing should be encapsulated in models and used in controllers._**
* [ ] **_View-related logic can be encapsulated in View helpers (app/helpers/\*_helper.rb)_**
* [ ] **_There should be absolutely NO logic in views._**

* [ ] **_Although the form_with is a good choice but the requirement is: "It has a sign up form created with #form_for in `app/views/users` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/sign_up_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49722686/))"_**

### Optional: you can use this :point_down: in the helper to check if user is signed in
```ruby
def signed_in?
  !current_user.nil?
end
```

Reply:
**A note for the next TSE** Dear TSE, we have made the first couple of changes requested by the previous TSE, however the latter requests to remove ALL logic from the views appears to be conflicting with the material we we're given from the ODIN project and other related material leading up to this project. The material states almost all logic should be handled by the controller, some logic related to showing and hiding elements on the page can still be kept in the views, which is what we did in our project. If this is a misunderstanding on our end, please point out how we should show/hide the elements otherwise. Thanks